File: JImageDisplay.java
package lab5;

import org.w3c.dom.css.RGBColor;

import javax.swing.*;
import java.awt.*;
import java.awt.image.BufferedImage;

public class JImageDisplay extends JComponent {
    public BufferedImage img;
    JImageDisplay(int width, int length){
       img = new BufferedImage(width,length,BufferedImage.TYPE_INT_RGB);
       Dimension dim = new Dimension(img.getWidth(),img.getHeight());
       setPreferredSize(dim);
    }

    protected void paintComponent(Graphics g){
    g.drawImage(img,0,0,img.getWidth(),img.getHeight(),null);
    }

    public void clearImage(){
        for (int i=0;i<img.getHeight();i++){
            for (int j=0;j<img.getWidth();j++){
                img.setRGB(i,j,0);
            }
        }
    }
    public void drawPixel(int x, int y,int rgbColor){
        img.setRGB(x,y,rgbColor);
    }
}

File: src/main/kotlin/org/dark0ghost/lab4/ui/component/JImageDisplay.kt
package org.dark0ghost.lab4.ui.component



import java.awt.Dimension
import java.awt.Graphics
import java.awt.image.BufferedImage
import java.awt.image.BufferedImage.TYPE_INT_RGB
import javax.swing.JComponent


open class JImageDisplay(height: UInt, width: UInt): JComponent() {
    private val bufferedImage: BufferedImage = BufferedImage(width.toInt(), height.toInt(), TYPE_INT_RGB)

    init {
        preferredSize = Dimension(width.toInt(), height.toInt())
        super.setPreferredSize(Dimension(width.toInt(), height.toInt()))
    }

    fun clearImage() {
        val blankArray = IntArray(width * height)
        bufferedImage.setRGB(0, 0, width, height, blankArray, 0, 1)
    }

    fun drawPixel(x: Int, y: Int, color: Int) = bufferedImage.setRGB(x, y, color)


    override fun paintComponent(g: Graphics) {
        super.paintComponent(g)
        g.drawImage(bufferedImage, 0, 0, bufferedImage.width, bufferedImage.height, null);
    }
}